src/other/codemirror/CodeMirror.tsx

```diff
@@ -69,17 +72,24 @@ class CodeMirror extends React.Component<CodeMirrorProps, CodeMirrorState> {
         const { content = "", options = {}, language, completions } = this.props;
         const completionExtension = autocompletion({ override: [completions] });
 
+        const { theme, onFocus, onBlur } = this.props;
         return (
             <CodeMirrorBase
                 value={content || ""}
                 // @ts-ignore
                 onChange={(editor: string, viewUpdate: { origin: string }) => {
                     this.handleContentChange(editor, viewUpdate);
                 }}
-                onFocus={() => this.setState({ isEditing: true })}
-                onBlur={() => this.setState({ isEditing: false })}
+                onFocus={() => {
+                    if (onFocus) onFocus();
+                    this.setState({ isEditing: true });
+                }}
+                onBlur={() => {
+                    if (onBlur) onBlur();
+                    this.setState({ isEditing: false });
+                }}
                 basicSetup={options}
-                theme="light"
+                theme={theme || "light"}
```

Comment:
Should such things be moved to defaultProps?

Reply:
This is OK atm